File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPoliticaBO/PoliticaPrestamoBO.cs
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftBibliotecaPoliticaPrestamos
{
    public class PoliticaPrestamoBO : BaseBO
    {
        public int insertar(int idMoraAsociada, int cantDiasPrestamoRegular,
                            int cantMaxCopiasPorDevolverPorUsuario, int cantDiasDeAmpliacionRegular, int cantMaxAmpliacionesPermitidasPorPrestamo,
                            int cantDiasParaDarPorPerdidaUnaCopia, int cantDiasSinPrestamoPorAtraso, int cantMaxAtrasosPorCiclo,
                            int cantDiasSinPrestamoPorMalEstado, float cargoPorMalEstado, int cantMaxMalEstadoPorCiclo,
                            int cantDiasSinPrestamoPorPerdido, float cargoPorPerdido, int cantMaxPerdidasPorCiclo)
        {
            return this.WsCliente.politicaprestamo_insertar(idMoraAsociada, cantDiasPrestamoRegular, cantMaxCopiasPorDevolverPorUsuario, cantDiasDeAmpliacionRegular, cantMaxAmpliacionesPermitidasPorPrestamo, cantDiasParaDarPorPerdidaUnaCopia, cantDiasSinPrestamoPorAtraso, cantMaxAtrasosPorCiclo, cantDiasSinPrestamoPorMalEstado, cargoPorMalEstado, cantMaxMalEstadoPorCiclo, cantDiasSinPrestamoPorPerdido, cargoPorPerdido, cantMaxPerdidasPorCiclo);
        }

        public int modificar(int idPoliticaPrestamo, int idMoraAsociada, int cantDiasPrestamoRegular,
                            int cantMaxCopiasPorDevolverPorUsuario, int cantDiasDeAmpliacionRegular, int cantMaxAmpliacionesPermitidasPorPrestamo,
                            int cantDiasParaDarPorPerdidaUnaCopia, int cantDiasSinPrestamoPorAtraso, int cantMaxAtrasosPorCiclo,
                            int cantDiasSinPrestamoPorMalEstado, float cargoPorMalEstado, int cantMaxMalEstadoPorCiclo,
                            int cantDiasSinPrestamoPorPerdido, float cargoPorPerdido, int cantMaxPerdidasPorCiclo)
        {
            return this.WsCliente.politicaprestamo_modificar(idPoliticaPrestamo, idMoraAsociada, cantDiasPrestamoRegular, cantMaxCopiasPorDevolverPorUsuario, cantDiasDeAmpliacionRegular, cantMaxAmpliacionesPermitidasPorPrestamo, cantDiasParaDarPorPerdidaUnaCopia, cantDiasSinPrestamoPorAtraso, cantMaxAtrasosPorCiclo, cantDiasSinPrestamoPorMalEstado, cargoPorMalEstado, cantMaxMalEstadoPorCiclo, cantDiasSinPrestamoPorPerdido, cargoPorPerdido, cantMaxPerdidasPorCiclo);
        }

        public int eliminar(int idPoliticaPrestamo)
        {
            return this.WsCliente.politicaprestamo_eliminar(idPoliticaPrestamo);
        }

        public BindingList<politicaPrestamo> listarTodos(int idMoraAsociada, int limite)
        {
            try
            {
                politicaPrestamo[] politicasPrestamo = this.WsCliente.politicaprestamo_listarTodos(idMoraAsociada, limite);
                if (politicasPrestamo == null) return new BindingList<politicaPrestamo>();
                return new BindingList<politicaPrestamo>(politicasPrestamo);
            }
            catch (Exception e)
            {
                return new BindingList<politicaPrestamo>();
            }
        }

        public politicaPrestamo obtenerPorId(int idPoliticaPrestamo)
        {
            return this.WsCliente.politicaprestamo_obtenerPorId(idPoliticaPrestamo);
        }
    }
}

File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/PoliticaPrestamoBO.java
package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import pe.edu.pucp.softbiblioteca.politicas.model.PoliticaPrestamo;
import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;

public interface PoliticaPrestamoBO extends Remote{
    
    public Integer insertar(Integer idMoraAsociada,Integer cantDiasPrestamoRegular,
                            Integer cantMaxCopiasPorDevolverPorUsuario,Integer cantDiasDeAmpliacionRegular,Integer cantMaxAmpliacionesPermitidasPorPrestamo,
                            Integer cantDiasParaDarPorPerdidaUnaCopia,Integer cantDiasSinPrestamoPorAtraso,Integer cantMaxAtrasosPorCiclo,
                            Integer cantDiasSinPrestamoPorMalEstado,Float cargoPorMalEstado,Integer cantMaxMalEstadoPorCiclo,
                            Integer cantDiasSinPrestamoPorPerdido,Float cargoPorPerdido,Integer cantMaxPerdidasPorCiclo) throws RemoteException;
    
    public Integer modificar(Integer idPoliticaPrestamo,Integer idMoraAsociada,Integer cantDiasPrestamoRegular,
                            Integer cantMaxCopiasPorDevolverPorUsuario,Integer cantDiasDeAmpliacionRegular,Integer cantMaxAmpliacionesPermitidasPorPrestamo,
                            Integer cantDiasParaDarPorPerdidaUnaCopia,Integer cantDiasSinPrestamoPorAtraso,Integer cantMaxAtrasosPorCiclo,
                            Integer cantDiasSinPrestamoPorMalEstado,Float cargoPorMalEstado,Integer cantMaxMalEstadoPorCiclo,
                            Integer cantDiasSinPrestamoPorPerdido,Float cargoPorPerdido,Integer cantMaxPerdidasPorCiclo) throws RemoteException;

    public Integer eliminar(Integer idPoliticaPrestamo) throws RemoteException;
    
    public ArrayList<PoliticaPrestamo> listarTodos(Integer idMoraAsociada, Integer limite) throws RemoteException;

    public PoliticaPrestamo obtenerPorId(Integer idPoliticaPrestamo) throws RemoteException;
    
}
